flytekit/remote/remote.py

```diff
@@ -1271,16 +1299,35 @@ def register_launch_plan(
                 source_root=project_root,
                 project=project or self.default_project,
                 domain=domain or self.default_domain,
+                version=version,
+            )
+
+        if self._wf_exists(
+            name=entity.workflow.name,
+            version=version,
+            project=serialization_settings.project,
+            domain=serialization_settings.domain,
+        ):
+            # Underlying workflow, exists, only register the launch plan itself
+            launch_plan_model = get_serializable(
+                OrderedDict(), settings=serialization_settings, entity=entity, options=options
+            )
+            ident = self.raw_register(
+                launch_plan_model, serialization_settings, version, create_default_launchplan=False
+            )
+            if ident is None:
+                raise ValueError("Failed to register launch plan")
```

Comment:
Should we raise this error in `self.raw_register`? Also, are we able to add the error message here? Something like `Failed to register launch plan with err: {err}`

Reply:
i say we keep this here until we can make the raw_register signature not an optional.